lib/CL/pocl_llvm_wg.cc

```diff
@@ -508,6 +508,8 @@ static void addStage1PassesToPipeline(cl_device_id Dev,
   // this must be done AFTER inlining, see note above
   addPass(Passes, "automatic-locals", PassType::Module);
 
+  addPass(Passes, "fix-unreachable");
```

Comment:
Let's avoid adding new small passes as I've just got rid of those in my private branch. Can you convert this to a helper function called from WILoops or similar? See the new DeSPMD/WILoops in my branch.

Reply:
> Can you convert this to a helper function called from WILoops or similar?

AFAICT that's not possible. Different parts of this "handling" of unreachables need to be done in different points of the pipeline. The first part (where functions with unreachable are marked inlined) must be done before AlwaysInline pass, the second part ("fixUnreachable" - where all unreachable instructions are removed) must be run before ImplicitConditionalBarriers; so AFAICT no, this cannot be done from WILoops.

All of the work of this pass is done in a single function "fixUnreachable", so in your private branch, should be easy to convert this pass to a function call in the right place.